symengine/polynomial_multivariate.cpp

```diff
@@ -1,220 +0,0 @@
-#include <symengine/polynomial_multivariate.h>
-
-namespace SymEngine
-{
-
-vec_basic MultivariateIntPolynomial::get_args() const
-{
-    vec_basic args;
-    umap_uvec_mpz d;
-    // To change the ordering in which the terms appear in the vector, use
-    // a different comparator for order_umap
-    auto v = sorted_keys(dict_);
-    for (const auto &p : v) {
-        map_basic_basic b;
-        int whichvar = 0;
-        for (auto sym : vars_) {
-            if (integer_class(0) != p[whichvar])
-                insert(b, sym, integer(p[whichvar]));
-            whichvar++;
-        }
-        args.push_back(
-            Mul::from_dict(integer(dict_.find(p)->second), std::move(b)));
-    }
-    return args;
-}
-
-std::size_t MultivariateIntPolynomial::__hash__() const
-{
-    std::size_t seed = MULTIVARIATEINTPOLYNOMIAL;
-    for (auto var : vars_)
-        hash_combine<std::string>(seed, var->__str__());
-
-    for (auto &p : dict_) {
-        std::size_t t = vec_uint_hash()(p.first);
-        hash_combine<std::size_t>(t, mp_get_si(p.second));
-        seed ^= t;
-    }
-    return seed;
-}
-
-int MultivariateIntPolynomial::compare(const Basic &o) const
-{
-    // copied from UnivariateIntPolynomial::compare and then modified.
-    const MultivariateIntPolynomial &s
-        = static_cast<const MultivariateIntPolynomial &>(o);
-
-    if (vars_.size() != s.vars_.size())
-        return vars_.size() < s.vars_.size() ? -1 : 1;
-    if (dict_.size() != s.dict_.size())
-        return dict_.size() < s.dict_.size() ? -1 : 1;
-
-    int cmp = unified_compare(vars_, s.vars_);
-    if (cmp != 0)
-        return cmp;
-
-    return unified_compare(dict_, s.dict_);
-}
-
-integer_class MultivariateIntPolynomial::eval(
-    std::map<RCP<const Basic>, integer_class, RCPBasicKeyLess> &vals) const
-{
-    integer_class ans(0);
-    for (auto bucket : dict_) {
-        integer_class term = bucket.second;
-        unsigned int whichvar = 0;
-        for (auto sym : vars_) {
-            integer_class temp;
-            mp_pow_ui(temp, vals.find(sym)->second, bucket.first[whichvar]);
-            term *= temp;
-            whichvar++;
-        }
-        ans += term;
-    }
-    return ans;
-}
-
-RCP<const MultivariateIntPolynomial>
-MultivariateIntPolynomial::convert(const UIntPoly &o)
```

Comment:
So, right now, I have removed the functionality for converting univariate polynomials to their multivariate counterparts. Which also made me change the `add_mpoly` etc. functions. Now, all we need is a general  purpose `add_poly` etc. functions, which do the "type" of poly check (multi/uni?) and their implementation (flint/piranha)? . Then the necessary conversions will will be done and the specific `add_mpoly` or `add_upoly` function will be called.